src/gsy_e/models/strategy/energy_parameters/virtual_heat_pump.py

```diff
@@ -0,0 +1,196 @@
+
+        self._dh_water_flow_m3: [DateTime, float] = EnergyProfile(
+            dh_water_flow_m3_profile, dh_water_flow_m3_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+        return {
+            **super().serialize(),
+            "water_supply_temp_C": self._water_supply_temp_C.input_profile,
+            "water_supply_temp_C_uuid": self._water_supply_temp_C.input_profile_uuid,
+            "water_return_temp_C": self._water_return_temp_C.input_profile,
+            "water_return_temp_C_uuid": self._water_return_temp_C.input_profile_uuid,
+            "dh_water_flow_m3_profile": self._dh_water_flow_m3.input_profile,
+            "dh_water_flow_m3_profile_uuid": self._dh_water_flow_m3.input_profile_uuid,
+        }
+
+    def _rotate_profiles(self, current_time_slot: Optional[DateTime] = None):
+        self._water_return_temp_C.read_or_rotate_profiles()
+        self._water_supply_temp_C.read_or_rotate_profiles()
+        self._dh_water_flow_m3.read_or_rotate_profiles()
+        self.state.delete_past_state_values(current_time_slot)
+
+    def _calc_energy_to_buy_maximum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(self._max_temp_C, time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
+
+    def _calc_energy_to_buy_minimum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(
+            self.state.get_storage_temp_C(time_slot), time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
+
+    def _calc_temp_decrease_K(self, time_slot: DateTime) -> float:
+        dh_supply_temp = self._water_supply_temp_C.profile[time_slot]
+        dh_return_temp = self._water_return_temp_C.profile[time_slot]
+        m_m3 = self._dh_water_flow_m3.profile[time_slot]
+        m_kg_per_sec = m_m3 * 1000 / 3600
+        q_out = m_kg_per_sec * WATER_SPECIFIC_TEMPERATURE * (dh_supply_temp - dh_return_temp)
+        temp_differential_per_sec = -q_out / (
+                WATER_DENSITY * WATER_SPECIFIC_TEMPERATURE * self._tank_volume_l)
+        temp_decrease = temp_differential_per_sec * GlobalConfig.slot_length.total_seconds()
+        assert temp_decrease <= 0.0
+        return abs(temp_decrease)
+
+    def _calc_temp_increase_K(self, time_slot: DateTime, energy_kWh: float) -> float:
+        storage_temp = self._energy_to_storage_temp(energy_kWh, time_slot)
+        current_storage_temp = self.state.get_storage_temp_C(time_slot)
+        if storage_temp <= current_storage_temp:
+            return 0
+        return storage_temp - current_storage_temp
+
+    def _storage_temp_to_energy(self, storage_temp: float, time_slot: DateTime) -> float:
```

Comment:
I think this method deserves a docstring even though it is private.
How about something like:
```python
"""
return the energy, that is needed to fill the heat demand which is equivalent to the temperature drop from supply to return temperature
"""
```
(did I get it right?)

Reply:
Also, how about renaming `storage_temp` into `target_storage_temp_C` ?
Maybe also the method name could include `target` ?

Reply:
I left a comment, this is actually the energy required to heat the water and bring the water tank to storage_temp degrees celcius